test/e2e/drain_tool_test.go

```diff
@@ -0,0 +1,283 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"go.goms.io/fleet/test/e2e/framework"
+)
+
+var _ = Describe("Drain cluster successfully", Ordered, Serial, func() {
+	crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
+	var drainClusters, noDrainClusters []*framework.Cluster
+	var noDrainClusterNames []string
+
+	BeforeAll(func() {
+		drainClusters = []*framework.Cluster{memberCluster1EastProd}
+		noDrainClusters = []*framework.Cluster{memberCluster2EastCanary, memberCluster3WestProd}
+		noDrainClusterNames = []string{memberCluster2EastCanaryName, memberCluster3WestProdName}
+
+		By("creating work resources")
+		createWorkResources()
+
+		// Create the CRP.
+		crp := &placementv1beta1.ClusterResourcePlacement{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: crpName,
+				// Add a custom finalizer; this would allow us to better observe
+				// the behavior of the controllers.
+				Finalizers: []string{customDeletionBlockerFinalizer},
+			},
+			Spec: placementv1beta1.ClusterResourcePlacementSpec{
+				Policy: &placementv1beta1.PlacementPolicy{
+					PlacementType: placementv1beta1.PickAllPlacementType,
+				},
+				ResourceSelectors: workResourceSelector(),
+			},
+		}
+		Expect(hubClient.Create(ctx, crp)).To(Succeed(), "Failed to create CRP %s", crpName)
+	})
+
+	AfterAll(func() {
+		// Uncordon member cluster 1 again to guarantee clean up of cordon taint on test failure.
+		runUncordonClusterBinary(hubClusterName, memberCluster1EastProdName)
```

Comment:
Hi Arvind! Just a nit: since the cordon binary itself is a test target, it might be best not to rely it on the cleanup part, just in case.

Reply:
Same applies for a few other test specs I assume.